proposals/views/proposal_views.py

```diff
@@ -357,6 +358,19 @@ def get_next_url(self):
     def get_back_url(self):
         """Return to the consent form overview of the last Study"""
         return reverse('proposals:consent', args=(self.object.pk,))
+    
+class ProposalUpdateDataManagement(braces.GroupRequiredMixin, generic.UpdateView):
+    """
+    Allows the secretary to change the Data Management Plan on the Proposal level
+    """
+    model = Proposal
+    template_name = 'studies/study_update_attachments.html'
+    form_class = ProposalUpdateDataManagementForm
+    group_required = settings.GROUP_SECRETARY
+
+    def get_success_url(self):
+        """Continue to the URL specified in the 'next' POST parameter"""
+        return self.request.POST.get('next', '/')
```

Comment:
So instead of just returning a URL to `ReviewDetailView` here, we retrieve a `next=` parameter POSTed from a hidden input in the template which generates the URL from the `HTTP_REFERER` header in the Django request object. All to end up pointing at the same `ReviewDetailView`.

9/10 I'll allow it.

Reply:
Ah classic case of: stole it from elsewhere meaning to update it, but it happened to already just work :-p I updated it for that 10/10 rating

Reply:
Yeah I see it comes from the UpdateAttachments page. Honestly it was just kind of funny but I'm glad you made it right.